builtin/aws/ecs/task.go

```diff
@@ -0,0 +1,249 @@
+source authentication information for AWS, using the configured task role.
+If no task role name is specified, Waypoint will create one with the required
+permissions.
+`)
+
+	doc.SetField(
+		"odr_task_role_name",
+		"Task role name to be used for the task role in the On-Demand Runner task",
+		docs.Summary(`
+This role must have the correct IAM policies to complete its task.
+If this IAM role does not already exist, a role will be created with the correct
+permissions"
+`,
+		),
+	)
+
+	return doc, nil
+}
+
+// TaskLauncher implements Configurable
+func (p *TaskLauncher) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// StopTask signals to docker to stop the container created previously. This
+// method is currently unimplemented.
+func (p *TaskLauncher) StopTask(
+	ctx context.Context,
+	log hclog.Logger,
+	ti *TaskInfo,
+) error {
+	log.Warn("the StopTask method is not currently implemented for ECS tasks")
+	return nil
+}
+
+// StartTask runs an ECS Task to perform the requested job.
+func (p *TaskLauncher) StartTask(
+	ctx context.Context,
+	log hclog.Logger,
+	tli *component.TaskLaunchInfo,
+) (*TaskInfo, error) {
+	// Generate an ID for our task name.
+	id, err := ulid.New(ulid.Now(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Arguments here represent the command inputs used when executing the
+	// waypoint runner, e.x:
+	//   "runner, agent, -vvv, -id, <some id>, -odr"
+	cmd := aws.StringSlice(tli.Arguments)
+
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: p.config.Region,
+	})
+	if err != nil {
+		return nil, err
+	}
+	ecsSvc := ecs.New(sess)
+
+	envs := []*ecs.KeyValuePair{}
+	for k, v := range tli.EnvironmentVariables {
+		envs = append(envs, &ecs.KeyValuePair{
+			Name:  aws.String(k),
+			Value: aws.String(v),
+		})
+	}
+	// This environment variable informs Kaniko that it is operating in a
+	// container. Without this environment variable, Kaniko will fail with an
+	// error
+	envs = append(envs, &ecs.KeyValuePair{
+		Name:  aws.String("container"),
+		Value: aws.String("docker"),
+	})
+
+	taskName := fmt.Sprintf("waypoint-odr-task-%s", id.String())
+
+	logOptions := buildLoggingOptions(
+		nil,
+		p.config.Region,
+		p.config.LogGroup,
+		taskName,
+	)
+
+	def := ecs.ContainerDefinition{
+		Name:        &taskName,
+		Image:       &tli.OciUrl,
+		Command:     cmd,
+		Environment: envs,
+		LogConfiguration: &ecs.LogConfiguration{
+			LogDriver: aws.String(ecs.LogDriverAwslogs),
+			Options:   logOptions,
+		},
+	}
+
+	exRoleArn, err := roleArn(p.config.ExecutionRoleName, sess)
+	if err != nil {
+		return nil, err
+	}
+
+	taskRoleArn, err := roleArn(p.config.TaskRoleName, sess)
+	if err != nil {
+		return nil, err
+	}
+
+	registerTaskDefinitionInput := ecs.RegisterTaskDefinitionInput{
+		ContainerDefinitions:    []*ecs.ContainerDefinition{&def},
+		ExecutionRoleArn:        &exRoleArn,
+		TaskRoleArn:             &taskRoleArn,
+		Cpu:                     aws.String("1024"),
```

Comment:
Might want to either use use the `p.config.{cpu,mem}` here, or make a new config entry for this. I can imagine someone needing higher values for complex build jobs.

Reply:
ODR memory and CPU are configurable as of c5c92919